### PR TITLE
Move the instance id generation logic into the vm_manager

### PR DESCRIPTION
**Pull request by andyliuliming**
why i'm doing is for add support for different type of the vm.
say, if we are going to use the vmss, the instance id logic is different with the vm.
so move the logic into the vm manager is better.

### CODE REVIEW

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb [hunk 1]
```diff
@@ -18,10 +18,16 @@ def initialize(azure_config, registry_endpoint, disk_manager, disk_manager2, azu
...
       # network_configurator contains service principal in azure_config so we must not log it.
-      @logger.info("create(#{instance_id}, #{location}, #{stemcell_id}, #{vm_props}, ..., ...)")
-      resource_group_name = instance_id.resource_group_name
+      @logger.info("create(#{bosh_vm_meta}, #{location}, #{vm_props}, ..., ...)")
+
+      instance_id = _build_instance_id(bosh_vm_meta, location, vm_props)
+      vm_params = {
+        instance_id: instance_id
+      }
+      # move the instance id creation logic into the create method.
```
Comment (gossion) [inline src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb]: remove this comment

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb [hunk 2]
```diff
@@ -83,13 +89,13 @@ def create(instance_id, location, stemcell_id, vm_props, network_configurator, e
       diagnostics_storage_account = task_get_diagnostics_storage_account.value!
 
       # create vm params
-      vm_params = {
+      vm_params.merge!(
```
Comment (gossion) [inline src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb]: weird to merge instance_id together with original vm_params but don't use it, see line 154.
And vm_params is for azure_client, and I suggest not to merge instance_id to it. If you want to use instance_id outside, just return two values: `return instance_id, vm_params`.

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb [hunk 3]
```diff
@@ -18,10 +18,16 @@ def initialize(azure_config, registry_endpoint, disk_manager, disk_manager2, azu
       @logger = Bosh::Clouds::Config.logger
     end
 
-    def create(instance_id, location, stemcell_id, vm_props, network_configurator, env)
+    def create(bosh_vm_meta, location, vm_props, network_configurator, env)
       # network_configurator contains service principal in azure_config so we must not log it.
-      @logger.info("create(#{instance_id}, #{location}, #{stemcell_id}, #{vm_props}, ..., ...)")
-      resource_group_name = instance_id.resource_group_name
+      @logger.info("create(#{bosh_vm_meta}, #{location}, #{vm_props}, ..., ...)")
```
Comment (gossion) [inline src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb]: a little misleading... suggest changing BoshVMMeta.to_s to
"<agent_id: #{@agent_id}, stemcell_id: #{@stemcell_id}>"

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb (hunk 2 above)
Comment (andyliuliming) [reply src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb]: yes, indeed.

File: src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb (hunk 3 above)
Comment (andyliuliming) [reply src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb]: but the vmmeta.to_s can be used in other places, so let's leave it be.

**Pull request merged**